osutil/stat.go

```diff
@@ -148,3 +148,20 @@ func ComparePathsByDeviceInode(a, b string) (match bool, err error) {
 
 	return os.SameFile(fi1, fi2), nil
 }
+
+// Major obtains the major device number.
+func Major(dev uint64) uint64 {
+	return ((dev >> 8) & 0xfff) | ((dev >> 32) & ^uint64(0xfff))
```

Comment:
While correct, it's harder to cross-reference with glibc headers. How about something like this:
```go
return (dev & 0x0000_0000_000f_f000) >> 8 | (dev & 0xffff_f000_0000_0000) >> 32 
```

This maps directly to glibc headers.

Reply:
Switching to unix.Mkdev/Major/Minor as hinted in https://github.com/canonical/snapd/pull/15212#discussion_r2044113652